docs/usage/annotations.rst

```diff
@@ -0,0 +1,118 @@
+Annotations
+===========
+
+A developer can create mappings between a concrete implementation (openqasm) and a program modelled in the `frontend <https://github.com/LEQO-Framework/low-code-modeler>`_ by using various openqasm3 annotations.
+
+Annotations can be emulated in openqasm2 by using special comments.
+
+.. warning::
+    The `whole line <https://openqasm.com/language/directives.html#annotations#:~:text=continue%20to%20the%20end%20of%20the%20line>`_ will be interpreted like an annotation.
+    Therefore you cannot use inline-comments on annotations!
+
+Input
+-----
+
+One input is defined as a single qubit register (:class:`~openqasm3.ast.QubitDeclaration`) with a single `@leqo.input` annotation.
+
+* Inputs can be of arbitrary size (See :ref:`input-memory-layout`)
+* Inputs have to be defined as contiguous memory
+    One :class:`~openqasm3.ast.QubitDeclaration` corresponds to one input
+* Input indices must be selected from a contiguous range of integers starting at `0`
+   No skips, no duplicates
+* Inputs may be declared anywhere in code
+* Input annotations may only appear on a :class:`~openqasm3.ast.QubitDeclaration`
+* Input annotations may only appear once per statement
+
+.. code-block:: openqasm3
+    :linenos:
+
+    // Qubit array
+    @leqo.input <<InputIndex>>
```

Comment:
For placeholders, it is usually a good idea to explicitly document what is part of the placeholder and how it should be replaced (i.e., replace `<<InputIndex>>` including the angled brackets with the index number of the input)